apps/web/src/components/Activity/utils.ts

```diff
@@ -0,0 +1,161 @@
+import { IntlShape, MessageDescriptor } from "react-intl";
+import PlusIcon from "@heroicons/react/16/solid/PlusIcon";
+import ArrowPathIcon from "@heroicons/react/16/solid/ArrowPathIcon";
+import TrashIcon from "@heroicons/react/16/solid/TrashIcon";
+import { tv, VariantProps } from "tailwind-variants";
+
+import { ActivityProperties, Maybe } from "@gc-digital-talent/graphql";
+import { IconType } from "@gc-digital-talent/ui";
+import { commonMessages } from "@gc-digital-talent/i18n";
+
+import activityMessages from "~/messages/activityMessages";
+
+export const icon = tv({
+  base: "mt-0.5 flex size-5.5 shrink-0 items-center rounded-full bg-primary-500 p-1 text-white",
+  variants: {
+    color: {
+      primary: "bg-primary-500",
+      success: "bg-success-500",
+      warning: "bg-warning-500",
+    },
+  },
+  defaultVariants: {
+    color: "primary",
+  },
+});
+
+export type IconVariants = VariantProps<typeof icon>;
+
+type ActivityEventType = "created" | "updated" | "added" | "deleted";
+
+export interface ActivityEventInfo {
+  message: MessageDescriptor;
+  icon: IconType;
+  color: IconVariants["color"];
+}
+
+function isEventType(value?: Maybe<string>): value is ActivityEventType {
+  return value === "created" || value === "updated";
+}
+
+const eventInfoMap = new Map<ActivityEventType, ActivityEventInfo>([
+  [
+    "created",
+    {
+      message: activityMessages.created,
+      icon: PlusIcon,
+      color: "primary",
+    },
+  ],
+  [
+    "added",
+    {
+      message: activityMessages.added,
+      icon: PlusIcon,
+      color: "primary",
+    },
+  ],
+  [
+    "updated",
+    {
+      message: activityMessages.updated,
+      icon: ArrowPathIcon,
+      color: "primary",
+    },
+  ],
+  [
+    "deleted",
+    {
+      message: activityMessages.deleted,
+      icon: TrashIcon,
+      color: "primary",
+    },
+  ],
+]);
+
+export function getEventInfo(
+  propsObj?: Maybe<ActivityProperties>,
+  event?: Maybe<string>,
+): ActivityEventInfo | undefined {
+  let eventType: ActivityEventType = "updated";
+  if (isEventType(event)) {
+    eventType = event;
+  }
+
+  if (propsObj && (!("old" in propsObj) || !propsObj.old)) {
+    eventType = "added";
+  }
```

Comment:
I think this may be a bit inaccurate for our json encode defaults
example `json_encode(['en' => '', 'fr' => ''])`

Because those will never start as null. I don't think that is something to worry too much about at the moment though. Since for things like pools, having all `pool.advertisment_location` touches be update operations isn't a big deal
Just something to consider